### PR TITLE
Disable 'from ROOT import *' in python3 (ROOT-8931)

### PR DESCRIPTION
**Pull request by daritter**
`from ROOT import *` leads to a segmentation violation when used with Python 3.6.

One could try fixing it instead of disabling it but the code which is used to install the lookup handler in the module `RootModule::SetRootLazyLookup` depends on internal CPython implementation details of the dict class which are not part of the public API. As a consequence keeping this alive will lead to very
fragile code, require continuous effort to adapt to internal changes and cause a lot of #ifdef handling. (as a matter of fact, Python 3.7 would probably already require new changes to this code already). 

Also there's a statement in the bug report that `from ROOT import *`  is broken in Python 3 so I don't understand why it's not disabled as it will only confuse users: https://sft.its.cern.ch/jira/browse/ROOT-8931

As such I would propose to instead have a clear error message that `from ROOT import *` does not work. This pr adds an `ImportError` which is easy to handle but cannot be just ignored by the user.

### CODE REVIEW

**Comment by Teemperor** [comment]
@phsft-bot build

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

File: bindings/pyroot/ROOT.py [hunk 1]
```diff
@@ -427,6 +427,8 @@ def __getattr1( self, name ):             # "start-up" getattr
    def __getattr2( self, name ):             # "running" getattr
     # handle "from ROOT import *" ... can be called multiple times
       if name == '__all__':
+         if sys.hexversion >= 0x3060000:
+            raise ImportError('"from ROOT import *" is not supported in Python 3')
```
Comment (etejedor) [inline bindings/pyroot/ROOT.py]: Hi Martin,
We should perhaps write "not supported in Python>=3.6" in the error message.

Comment (daritter) [reply bindings/pyroot/ROOT.py]: As the statement in the JIRA issue was that it "does not work [reliably] in python 3" I chose to be more general in the error message. Apart from us I don't think there are many people using ROOT and python3 yet. But I have no problem to change this to the more specific message you suggest.

Comment (etejedor) [reply bindings/pyroot/ROOT.py]: We can disable it for all Python3, but then the condition check needs to be modified accordingly. Thanks Martin!

**Comment by Axel-Naumann** [comment]
@etejedor ready to merge?

**Comment by etejedor** [comment]
@Axel-Naumann yes!

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

**Pull request merged**